core/src/main/java/org/apache/accumulo/core/conf/PropertyType.java

```diff
@@ -186,6 +195,39 @@ public boolean isValidFormat(String value) {
     return predicate.test(value);
   }
 
+  /**
+   * Validate that the provided string can be parsed into a json object. This implementation uses
+   * jackson databind because it is less permissive that GSON for what is considered valid. This
+   * implementation cannot guarantee that the json is valid for the target usage. That would require
+   * something like a json schema or a check specific to the use-case. This is only trying to
+   * provide a generic, minimal check that at least the json is valid.
+   */
+  private static class ValidJson implements Predicate<String> {
+    private static final Logger log = LoggerFactory.getLogger(ValidJson.class);
+
+    // set a limit of 1 million characters on the string as rough sanity check
+    private static final int ONE_MILLION = 1024 * 1024;
+
+    @Override
+    public boolean test(String value) {
+      try {
+        if (value.length() > ONE_MILLION) {
+          log.info("provided json string length {} is greater than limit of {} for parsing",
+              value.length(), ONE_MILLION);
+          return false;
+        }
+        ObjectMapper mapper =
+            new ObjectMapper().enable(DeserializationFeature.FAIL_ON_READING_DUP_TREE_KEY)
+                .enable(DeserializationFeature.FAIL_ON_TRAILING_TOKENS);
```

Comment:
Similar comment here. Not sure how expensive this object is to create but maybe a `private static final ObjectMapper` could be created within this class for reuse.

Reply:
Made static in 7fa8acabb7.  Also, left documentation that we could consider using ThreadLocal.  The ObjectMapper is thread safe, but apparently uses synchronization, so if we see contention then there are alternatives.

Reply:
Unnecessary. Just throw it inside the ValidJson class as a private instance field inside that. See my previous comment about passing `ValidJson::new` as the predicate.

Reply:
Changed in 33bc47d2a1